hbase-common/src/main/java/org/apache/hadoop/hbase/util/UnsafeAccess.java

```diff
@@ -471,4 +485,57 @@ public static byte toByte(ByteBuffer buf, int offset) {
   public static byte toByte(Object ref, long offset) {
     return theUnsafe.getByte(ref, offset);
   }
+
+  /**
+   * Zero fill a byte buffer as efficiently as possible.
+   * @param buf the byte buffer
+   */
+  public static void clear(final ByteBuffer buf) {
+    if (buf.isDirect()) {
+      clear(((DirectBuffer)buf).address(), buf.capacity());
```

Comment:
I think we could just use Unsafe.setMemory in this case, it should just be a memset call in the native code I suppose.

Reply:
memset doesn't use an 8-byte stride, that is why my version is better. I suppose some memsets in some c libraries might try to optimize, but we can't depend on that.

Reply:
OK, I checked the source code of openjdk, it is not a memset, it will try to use long if the address and length are both 8 bytes aligned. It is a bit strange that they do not use the typical loop unrolling way to implement this function, as what you do here.

https://github.com/adoptium/jdk11u/blob/c9c728849c137e21535cc402664dd54ee421b0d3/src/hotspot/share/utilities/copy.cpp#L238

And the behavior is also well documented in the javadoc of the Unsafe.setMemory method.
```
     * <p>The stores are in coherent (atomic) units of a size determined
     * by the address and length parameters.  If the effective address and
     * length are all even modulo 8, the stores take place in 'long' units.
     * If the effective address and length are (resp.) even modulo 4 or 2,
     * the stores take place in units of 'int' or 'short'.
```

So let's keep your version here. And could also add some comments to say why we do not use Unsafe.setMemory.

Reply:
I will change the patch to use Unsafe.setMemory @Apache9 . I think they do the alignment test because some architectures do not support unaligned stores. x86 does. Not sure about ARM. I will change this so we might avoid a bug report in the future for segfault due to unaligned access on RISC-V or something.

Reply:
Yes, unaligned stores will be a problem for some architectures, and for most architectures, aligned stores will be faster than unaligned stores. But usually, the address should be aligned as the compiler will do padding, and it is a bit strange that the implementation will skip 8 bytes or 4 bytes if the length is not 8 bytes aligned.

I suppose the implementation should first do some stores to make the later stores aligned, and then do the main loops with 8 bytes aligned(or maybe more if we MMX or SSE), and finally we deal with the remaining several bytes if needed.

Anyway, this could an optimization later, not a blocker issue for now.